sktime/transformations/panel/augmenter.py

```diff
@@ -0,0 +1,1217 @@
+        See the documentation of the specific augmenter for details.
+    use_relative_fit: bool, optional (default = False)
+        Whether to a fit the augmenter regarding a statistical parameter of
+        the data. If True, relative_fit_stat_fun specifies the statistical
+        parameter and relative_fit_type specifies the fitting reference data.
+    relative_fit_stat_fun: a function, optional (default = np.std)
+        Specifies the statistical parameter to calculate. Any function
+        that satisfies the following syntax is allowed: stat = fun(X:
+        pd.Series). The only allowed input is a numeric pd.Series and output
+        has to be float or int. The default is the standard deviation np.std.
+    relative_fit_type: str, optional (default = "fit")
+        "fit": fit demanded statistics with a separate train set.
+        "fit-transform": fit statistics just before transformation
+        regarding the whole given panel.
+        "instance-wise": fit statistics just before transformation
+        individually per instance (and variable).
+    random_state: int, optional (default = None)
+        A random state seed for reproducibility.
+    excluded_var_indices: iterable of int optional (default = None)
+        Iterable (e.g. tuple or list) of int, containing the indices of those
+        variables to exclude from augmentation. Default is None and all
+        variables are used.
+    n_jobs: int, optional (default = 1)
+        Integer specifying the maximum number of concurrently running
+        workers on specific panel's instances. If 1 is given, no parallelism is
+        used. If set to -1, all CPUs are used. For n_jobs below -1, (n_cpus
+        + 1 + n_jobs) are used. For example with n_jobs=-2, all CPUs but one
+        are used. Not implemented yet.
+    """
+
+    _tags = {
+        "scitype:transform-input": "Panel",
+        "scitype:transform-output": "Panel",
+        "scitype:transform-labels": "None",
+        "scitype:instancewise": True,
+        "univariate-only": False,
+        "handles-missing-data": False,
+        "X_inner_mtype": "nested_univ",
+        "y_inner_mtype": "pd.Series",
+        "X-y-must-have-same-index": False,
+        "enforce_index_type": None,
+        "fit-in-transform": False,
+        "transform-returns-same-time-index": False,
+        "skip-inverse-transform": True,
+    }
+
+    def __init__(
+        self,
+        p: float = 1.0,
+        param=None,
+        use_relative_fit=False,
+        relative_fit_stat_fun=np.std,
+        relative_fit_type="fit",
+        random_state=None,
+        excluded_var_indices=None,
+        n_jobs=1,
+    ):
+        # input parameters
+        self.p = p
+        self.param = param
+        self.use_relative_fit = use_relative_fit
+        self.relative_fit_stat_fun = relative_fit_stat_fun
+        self.relative_fit_type = relative_fit_type
+        self.random_state = random_state
+        if excluded_var_indices is None:
```

Comment:
If you would like to stay compatible with sklearn design principles (https://scikit-learn.org/stable/developers/develop.html#apis-of-scikit-learn-objects) there should not be any logic in the `__init__` methods beyond setting the attributes.

Reply:
Thanks for the link, That should be possible by moving all checks, etc. to `_fit()` (seems also to be the preferred way in sklearn). I will provide a updated solution soon.

Reply:
I fixed that.